packages/ui/src/components/ui-prompt.tsx

```diff
@@ -11,48 +11,55 @@ import {
 } from '@workspace/ui/components/dialog'
 import { Input } from '@workspace/ui/components/input'
 import { Label } from '@workspace/ui/components/label'
-import type { FormEvent, ReactNode } from 'react'
-import { useId, useState } from 'react'
+import type { ComponentProps, FormEvent, ReactNode } from 'react'
+import { useEffect, useId, useState } from 'react'
 
 interface UiPromptProps {
-  title: ReactNode
-  description: ReactNode
-  label: ReactNode
-  placeholder?: string
-  actionLabel: ReactNode
   action: (value: string) => void
+  actionLabel: ReactNode
   children: ReactNode
+  description?: ReactNode | undefined
+  inputProps?: ComponentProps<typeof Input>
+  label: ReactNode
+  placeholder?: string
+  title: ReactNode
+  value: string
 }
 
-export function UiPrompt({ title, description, label, placeholder, actionLabel, action, children }: UiPromptProps) {
+export function UiPrompt({
+  action,
+  actionLabel,
+  children,
+  description,
+  inputProps,
+  label,
+  placeholder,
+  title,
+  value: inputValue,
+}: UiPromptProps) {
   const [isOpen, setIsOpen] = useState(false)
-  const [value, setValue] = useState('')
+  const [value, setValue] = useState(inputValue)
   const inputId = useId()
 
-  function handleOpenChange(shouldBeOpen: boolean) {
-    setIsOpen(shouldBeOpen)
-    if (!shouldBeOpen) {
-      setValue('')
-    }
-  }
+  useEffect(() => {
```

Comment:
Any suggestions on how to do this without using `useEffect` is welcome

Reply:
I'm not sure if you can. Since this is a form I would probably have used react-hook-form with `values` prop if we need something like `inputValue` https://react-hook-form.com/docs/useform#values

Reply:
Added that to #270